cartography/intel/gsuite/api.py

```diff
@@ -0,0 +1,243 @@
+import logging
+
+from googleapiclient.discovery import HttpError
+
+
+logger = logging.getLogger(__name__)
+
+
+def get_all_groups(admin):
```

Comment:
Consider having some of the `get_*` functions return the raw response objects.  If I recall correctly, `response[id]` in the Google APIs is a helpful prefix that I've used in the past to uniquely identify nodes.  See https://github.com/lyft/cartography/blob/master/cartography/intel/gcp/compute.py#L141

On the other hand, since you use a group email later on to enumerate users, keeping the return type here the same can make sense.

In general though I'd recommend having a `get_*` in the Google API family return response objects and then have the transform functions massage them to usable objects for ingestion.

Reply:
I can do that, but as a consideration it'll add an additional for loop to transform the objects.

Reply:
Before I apply this to users workflow as well, here's how it looks like for `groups`
https://github.com/lyft/cartography/pull/126/commits/9afebd1a2f28bf02e69067b0e35f3250ff1427b4#diff-732718514a18eeb9183d728497f14713R34-R39

I'm not entirely convinced this would be the best course of action because we're essentially passing around large lists for transformation and adding additional loops.  However, if this is the workflow we want, i'll go ahead and apply.